config/sources/families/thead.conf

```diff
@@ -7,32 +7,55 @@
 # https://github.com/armbian/build/
 #
 
-ARCH="riscv64"
+# Source riscv64 generic configuration
+. "$SRC/config/sources/families/include/riscv64_common.inc"
 
-BOOTSOURCE='https://github.com/chainsx/thead-u-boot'
-BOOTBRANCH='branch:extlinux'
-BOOTPATCHDIR="legacy"
-UBOOT_TARGET_MAP=";;u-boot-with-spl.bin"
+# Generic configuration for the T-Head family
 
 LINUXFAMILY="thead"
 
-case "${BRANCH}" in
+# OpenSBI: Supervisor Binary Interface Firmware
+[ -n "$OPENSBISOURCE" ] || OPENSBISOURCE="https://github.com/riscv-software-src/opensbi"
+[ -n "$OPENSBIBRANCH" ] || OPENSBIBRANCH="branch:master"
 
-	legacy)
-		UBOOT_COMPILER="riscv64-linux-gnu-"
-		UBOOT_USE_GCC='< 8.0'
-		BOOTDIR='u-boot-thead'
-		KERNELSOURCE='https://github.com/revyos/thead-kernel'
-		KERNELBRANCH="branch:lpi4a"
-		declare -g KERNEL_MAJOR_MINOR="5.10"
-		KERNELPATCHDIR="thead-${BRANCH}"
-		LINUXCONFIG="linux-thead-${BRANCH}"
+OPENSBIDIR="opensbi"
 
+case "$BRANCH" in
+	"legacy"|"current")
+		[ -n "$OPENSBI_USE_GCC" ] || OPENSBI_USE_GCC="> 6.3"
+		OPENSBI_TARGET_MAP=";;build/platform/generic/firmware/fw_dynamic.bin"
 		;;
+	*) exit_with_error "OpenSBI is not implemented for release '$BRANCH'"
+esac
 
+# U-Boot: Bootloader
+[ -n "$BOOTSOURCE" ] || BOOTSOURCE="https://github.com/u-boot/u-boot"
+[ -n "$BOOTBRANCH" ] || BOOTBRANCH="branch:master"
+[ -n "$UBOOT_COMPILER" ] || UBOOT_COMPILER="riscv64-linux-gnu-"
+[ -n "$UBOOT_TARGET_MAP" ] || UBOOT_TARGET_MAP=";;u-boot-with-spl.bin"
+
+BOOTDIR="u-boot-thead"
+
+case "$BRANCH" in
+	legacy)
+		UBOOT_USE_GCC='< 8.0'
+		;;
+	*) display_alert "Building U-Boot on branch '$BRANCH' for board '$BOARD' is not implemented, success is not guaranteed" "warn"
 esac
 
+# Linux Kernel
+[ -n "$LINUXCONFIG" ] || LINUXCONFIG="linux-thead-$BRANCH"
+
+KERNELPATCHDIR="thead-$BRANCH"
+
+# Power Management
+GOVERNOR=performance
+# FIXME(Krey): Figure out sane values
+## [ -n $CPUMIN ] || CPUMIN=480000
+## [ -n $CPUMAX ] || CPUMAX=1010000
```

Comment:
Brainstorm? I dunno what values to put there as the default

Reply:
You can just don't assign values. Or take it from `/sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies`

Reply:
Will investigate..